server/src/zmq_worker.rs

```diff
@@ -304,38 +304,41 @@ async fn handle_incoming_request(
                         .context("ZmqWorker error while forwarding request to load balancer.")
                 }
                 _ => {
-                    error!(
-                        "ZmqWorker received a request with api {:?} while I can only handle \
-                    Status/Metadatas/PtPlanner/PlacesNearby/NextDepartures/NextArrivals api.",
-                        requested_api
-                    );
-                    Ok(())
+                    let err = format_err!("ZmqWorker received a request with api {requested_api:?} while I can only handle \
+                    Status/Metadatas/PtPlanner/PlacesNearby/NextDepartures/NextArrivals api.");
+                    error!("{:?}", err);
+                    // let's send back a response to our zmq client that we received an invalid query
+                    send_error_response(zmq_socket, request_message.client_id, &err).await
                 }
             }
         }
         Err(err) => {
-            let err_str = format!("Could not decode zmq message into protobuf. {:?}", err);
-            error!("{}", err_str);
-
+            let err = format_err!(err).context("Could not decode zmq message into protobuf.");
+            error!("{:?}", err);
```

Comment:
Can you do this? (not sure since this syntax might only by supported in standard library macros... :shrug: )
```suggestion
            error!("{err:?}");
```

Reply:
:+1: yes it works !